src/Delegation.sol

```diff
@@ -201,6 +204,14 @@ contract Delegation is IDelegation, EIP712, GuardedExecutor {
     /// This constant is a pun for "chain ID 0".
     uint16 public constant MULTICHAIN_NONCE_PREFIX = 0xc1d0;
 
+    /// @dev A unique identifier to be passed into `upgradeHook(bytes32 previousVersion)`
+    /// via the transient storage slot at `_UPGRADE_HOOK_GUARD_TRANSIENT_SLOT`.
+    bytes32 internal constant _UPGRADE_HOOK_ID = keccak256("PORTO_DELEGATION_UPGRADE_HOOK_ID");
+
+    /// @dev This transient slot must be set to `_UPGRADE_HOOK_ID` before `upgradeHook` can be processed.
+    bytes32 internal constant _UPGRADE_HOOK_GUARD_TRANSIENT_SLOT =
+        bytes32(uint256(keccak256("_UPGRADE_HOOK_GUARD_TRANSIENT_SLOT")) - 1);
```

Comment:
Trivia: this may not be actually evaluated on compile time (yes, it's so dumb).

But since the upgrade function is not often called, i think we can deal with some more readable Solidity instead of a hardcoded constant.